docs/design.md

```diff
@@ -104,3 +104,41 @@ several ways according to policy:
 
 Then like the usual LibOS pattern it will call the container's entry point as though the container
 was a process unto itself (rather than where it now sits like a library in the LibOS PAL process).
+
+## [Future Work] Local Attestation for Filesystem Decryption Key Provisioning
+
+![EnclaveCC_LA.png](images/EnclaveCC_LA.png)
+
+A local attestation (LA) feature is needed for authentication/authorization and secret provisioning
+between an Agent-enclave instance and a set of App-enclave instances resided in a same Pod. this
+design is a guide for enclave-CC developers to implement this feature.
+
+In this design, the authentication and authorization could be safely simplified in the context of
+enclave-CC by verifying and validating the SGX local report with customizable policy rules, e.g.,
+matching `mrenclave/mrsigner` combined with `configid`.
+
+In most scenarios, it is one-shot transactions used for single filesystem decryption key (FDK)
+provisioning. However, this LA protocol can easily be extended to provision multiple keys for more
+general purposes.
+
+To achieve this, it is necessary to add a key id to the second request message sent along G_a and
+app's report, with the ID being used to indicate which key is being requested. But whether the key
+id can be automatically enumerated or not is decided by the caller `ttrpc` service/client and their
+contexts. At this moment, the caller is not discussed yet. We hope to merge the protocol definition
+completed prior to the implementation. By then, we could make the final decision to add the key id
+into protocol or not.
+
+In addition to adding this ID, the generated nonce and derived `AEAD` keyset expected to be cached
+on both sides to keep the secure session alive which has been established in initial transaction.
+The session's security context will hold those cached values to ensure AIC triad and freshness for
+subsequent key provisioning requests.
+
+Furthermore, agent or app. should set the expiration time of its own security context properly to
+minimize risk exposure. Upon closing of underlying `ttrpc` connection, the security context will be
+cleared accordingly.
+
+Please note that there is no secure transport configured between `ttrpc` endpoints in our case,
+therefore it is necessary to check the integrity and ensure the confidentiality if any of whatever
+they received (e.g., `AEAD`, https://datatracker.ietf.org/doc/html/rfc5116). The `ttrpc` service
+needs to be strictly configured to listen on loopback interface only for security. In addition, the
+identity of app. could also be verified to ensure the provenance of the app. enclave.
```

Comment:
In this last sentence, I think 'app' and 'agent' got swapped on your latest revision.

But probably this all should be introduced early on and explicitly...
In your initial paragraphs you probably want to say that the agent holds a secret (FDK) and the app wants to authenticate to the agent in order for the agent to release the FDK. The app will prove its identity with a report and the agent will check that against a policy which requires a matching MRSIGNER or MRENCLAVE, possibly including an additional value from KSS. This policy is also really the heart of what this design should explain for enclave-cc.
(i.e. Local attestation as a generic protocol is shown in the linux-sgx examples and the SGX documentation, so discussing things specific to enclave-cc is most helpful for enclave-cc documentation and saves you trouble of rewriting things that may be documented elsewhere. 
Similarly the AEAD algorithm selection should be specified, but if the thinking is just to build on the linux-sgx example it might be better to just reference that.).


In this last sentence I think you meant to offer that the App could also authenticate the Agent. If you want to suggest that, please offer some discussion why this would be beneficial. The only things that come to mind for me are availability attacks which are outside the CC model.
Please also update the diagram to mark the app checking the agent against policy as optional or consider dropping that part.

Reply:
Thank you for the comments, I have updated the doc and diagram accordingly.